src/components/common/applicant-info-1.vue

```diff
@@ -550,6 +558,20 @@ export default class ApplicantInfo1 extends Mixins(ActionMixin) {
     return this.getNrData?.xproJurisdiction
   }
 
+  handleNameBlur (key, event) {
```

Comment:
This updates the local (class) properties but not the store properties.

The store properties are updated by `updateApplicant`.

And, you also bind the properties to the v-model prop (which is equivalent to a `value` prop and an `input` event) - see https://vuejs.org/v2/guide/forms.html#Basic-Usage

This is too complicated.

Reply:
updateApplicant does not update the trimmed name in UI. I think it is better to let user see the formated/trimed input string after the input finished.

Reply:
I've updated code by calling setApplicantDetails instead of using v-model. Please review again.

Reply:
Your changes look much better!

I think you still have a conflict or duplication with `@blur` and `@input`, which both call setApplicantDetails().

Reply:
I un-resolved this conversation. Since I started this conversation, let me close it. I use it as a reminder to look at the changes surrounding this conversation. (If you close it then I might lose track of outstanding issues.)

Reply:
> I think you still have a conflict or duplication with @blur and @input, which both call setApplicantDetails().

Reply:
removeExcessSpace can only be called when input finished (@blur). @blur has to call setApplicationDetails to update the trimmed input text in UI.
Because the function called by @input is a common function which shared by all the text input in this page, so far it is good. I don't want to change it.
Your suggestions?

Reply:
What do you think of this (for the 3 name fields only)?
```
     ...
     @blur="handleNameBlur('firstName', $event)"
     @input="this.clearValidation()"
     ...
```

Reply:
is just using `@onblur` problematic?  i.e. are there scenarios where just using `@onblur` will produce undesirable outcomes?

Reply:
> What do you think of this (for the 3 name fields only)?
> 
> ```
>      ...
>      @blur="handleNameBlur('firstName', $event)"
>      @input="this.clearValidation()"
>      ...
> ```

I tested after changing @input function as your suggested. However, not all the test cases are successful: the input value wasn't updated after trimming. I am not sure why.

Reply:
> Currently, clearValidation() is called @input. It may not be a big problem call it in @blur, but I don't want to change existing functionalities.

I don't know. Try it and test it, I guess.

Reply:
There something going on in the store actions/mutate code that requires both the blur and input events to be handled in order to be reactive. I don't understand it, and this was supposed to be a quick fix... so, let's go with this code after all.

To mitigate another developer wasting time trying to figure this out, please add a comment (eg, to handleNameBlur) to describe what you found here.

Also, I'm not sure why GH isn't flagging this as a conflict, but the app version in the package files has been updated (by a high priority fix). Please rebase and update to the next version number.

After this, I will approve and merge this PR.

Reply:
Good comments - thanks, resolving this conversation.

Reply:
The version number updated and the comments added.